File: zjazd_3/app/src/main/java/com/example/zjazd_2/MainActivity.java
package com.example.zjazd_2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button button_BMI;
    private Button button_callory;
    private Button button_przepis;

    private Button button_quiz;

    private Button button_chart;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button_BMI = (Button) findViewById(R.id.button_BMI);
        button_callory = (Button) findViewById(R.id.button_callory);
        button_przepis = (Button) findViewById(R.id.button_przepis);
        button_quiz = (Button) findViewById(R.id.button_quiz);
        button_chart = (Button) findViewById(R.id.button_chart);



        button_BMI.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openActivity1();
            }
        });

        button_callory.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openActivity2();
            }
        });
        button_przepis.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openActivity3();
            }
        });
        button_quiz.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openQuiz();
            }
        });
        button_chart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openChart();
            }
        });
    }

    public void openActivity1() {
        Intent intent = new Intent(this, Activity1.class);
        startActivity(intent);
    }
    public void openActivity2() {
        Intent intent = new Intent(this, Activity2.class);
        startActivity(intent);
    }
    public void openActivity3() {
        Intent intent = new Intent(this, Activity3.class);
        startActivity(intent);
    }

    public void openQuiz() {
        Intent intent = new Intent(this, Quiz.class);
        startActivity(intent);
    }

    public void openChart() {
        Intent intent = new Intent(this, Chart.class);
        startActivity(intent);
    }
}

File: zjazd_4/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView

import android.widget.Button


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val button:Button = findViewById(R.id.button)
        val weight: TextView = findViewById(R.id.Weight)
        val height: TextView = findViewById(R.id.Height)
        val text: TextView = findViewById(R.id.textView);



        button.setOnClickListener{
            val weight_value = weight.text.toString().toFloat()
            val height_value = height.text.toString().toFloat()

            val result =  (weight_value/(height_value*2))*100;
            text.setText(result.toString())

        }
    }
}